mindmeld/auto_annotator.py

```diff
@@ -152,66 +179,37 @@ def valid_entity_check(self, entity):
         entity = entity.lower().strip()
         return entity in self.supported_entity_types
 
-    def annotate(self, **kwargs):
-        """ Annotate data based on configurations in the config.py file.
-
-        Args:
-            kwargs (dict, optional): Configuration overrides can be passed in as arguments.
-        """
-        config = deepcopy(self.config)
-        for key, value in kwargs.items():
-            config[key] = value
-        if not config["annotate"]:
+    def annotate(self):
+        """Annotate data."""
+        if not self.annotation_rules:
             logger.warning(
                 """'annotate' field is not configured or misconfigured in the `config.py`.
                  We can't find any file to annotate."""
             )
             return
-        file_entities_map = self._get_file_entities_map(
-            action=AnnotatorAction.ANNOTATE, config=config
-        )
-        self._modify_queries(
-            file_entities_map, action=AnnotatorAction.ANNOTATE, config=config
-        )
-
-    def unannotate(self, **kwargs):
-        """ Unannotate data based on configurations in the config.py file.
-
-        Args:
-            kwargs (dict, optional): Configuration overrides can be passed in as arguments.
-            unannotate_all (bool): Unannotate all entities.
-        """
-        config = deepcopy(self.config)
-        for key, value in kwargs.items():
-            if key == "unannotate_all" and value:
-                config["unannotate"] = [
-                    {"domains": ".*", "intents": ".*", "files": ".*", "entities": ".*",}
-                ]
-                config["unannotate_supported_entities_only"] = False
-            else:
-                config[key] = value
+        file_entities_map = self._get_file_entities_map(action=AnnotatorAction.ANNOTATE)
+        self._modify_queries(file_entities_map, action=AnnotatorAction.ANNOTATE)
 
-        if not config["unannotate"]:
+    def unannotate(self):
+        """Unannotate data."""
+        if not self.unannotate:
             logger.warning(
                 """'unannotate' field is not configured or misconfigured in the `config.py`.
                  We can't find any file to unannotate."""
             )
             return
         file_entities_map = self._get_file_entities_map(
-            action=AnnotatorAction.UNANNOTATE, config=config
-        )
-        self._modify_queries(
-            file_entities_map, action=AnnotatorAction.UNANNOTATE, config=config
+            action=AnnotatorAction.UNANNOTATE
         )
+        self._modify_queries(file_entities_map, action=AnnotatorAction.UNANNOTATE)
```

Comment:
You can create a separate helper function as follows:
```python
def update_query_with_action(self, action=AnnotatorAction.UNANNOTATE);
     file_entities_map = self._get_file_entities_map(action)
    self._modify_queries(file_entities_map, action)
```
and then call that from the `annotate ` and `unannotate ` methods

Reply:
Moved "file_entities_map = self._get_file_entities_map(action)" into the "self._modify_queries()" helper function since this is shared code between the only two callers of this function.